### PR TITLE
Product image seo

### LINKED ISSUE #713: Structured data for product image missing
**Describe the enhancement you'd like**
After launching a Dawn theme, Google Search Console does not find product images anymore. Looking into the source code of the live page, I can confirm. There is code in the theme to add image structured data - but it doesn't seem to work. This happens for all products that don't have an explicit variant set up. It _does_ work for those products that have several variants.

**Additional context/screenshots**
Live page: https://www.frenchgirlorganics.com/products/lumiere-caramel-shimmer-oil
Google Results: https://search.google.com/test/rich-results?id=zOC-sJXgkJs2wYYIJGeyPg

Screenshot 2021-09-29 at 16 56 38
Screenshot 2021-09-29 at 16 55 41
Screenshot 2021-09-29 at 16 55 26
Screenshot 2021-09-29 at 16 55 14

### PR DESCRIPTION
**Pull request by ludoboludo**
**Why are these changes introduced?**

Fixes #713 

**What approach did you take?**

I added another option for when the product as a featured image but no variant image.

**Other considerations**

I wonder if we should have those two options or only use the `featured_media` one. 

As a user/merchant, my featured image for the product might be the one I'd like to use/show for general information while the variant image not as much.  

So I think using only `featured_media` would be best 🤔  Curious to hear reviewers' thoughts (cc: @bredowmax)

**Demo links**

- ~Store~
- In order to test you can copy the changes on your/a test store that isn't password protected. Edit the code of the theme (on `main-product.liquid` and/or `featured-product.liquid`), then grab the url of the home page if you're testing the feat. product or the product page if you're testing the main product and paste it in [google's rich results test](https://search.google.com/test/rich-results). 

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Review by KaichenWang — DISMISSED** [review]

File: sections/featured-product.liquid [hunk 1]
```diff
@@ -402,16 +402,24 @@
...
     "@type": "Product",
     "name": {{ product.title | json }},
     "url": {{ shop.url | append: product.url | json }},
-    {%- if product.selected_or_first_available_variant.featured_media -%}
-      {%- assign media_size = product.selected_or_first_available_variant.featured_media.preview_image.width | append: 'x' -%}
+    {%- if seoMedia -%}
+      {%- assign media_size = seoMedia.preview_image.width | append: 'x' -%}
       "image": [
-        {{ product.featured_media | img_url: media_size | prepend: "https:" | json }}
+        {{ seoMedia | img_url: media_size | prepend: "https:" | json }}
```
Comment (sofiamatulis) [inline sections/featured-product.liquid:422]:
```suggestion
        {{ seo_media | img_url: media_size | prepend: "https:" | json }}
```

File: sections/featured-product.liquid [hunk 2]
```diff
@@ -402,16 +402,24 @@
...
 <script type="application/ld+json">
   {
     "@context": "http://schema.org/",
     "@type": "Product",
     "name": {{ product.title | json }},
     "url": {{ shop.url | append: product.url | json }},
-    {%- if product.selected_or_first_available_variant.featured_media -%}
-      {%- assign media_size = product.selected_or_first_available_variant.featured_media.preview_image.width | append: 'x' -%}
+    {%- if seoMedia -%}
+      {%- assign media_size = seoMedia.preview_image.width | append: 'x' -%}
```
Comment (sofiamatulis) [inline sections/featured-product.liquid:420]:
```suggestion
      {%- assign media_size = seo_media.preview_image.width | append: 'x' -%}
```

File: sections/main-product.liquid [hunk 3]
```diff
@@ -476,16 +476,24 @@
   <script src="{{ 'product-model.js' | asset_url }}" defer></script>
 {%- endif -%}
 
+{%- liquid
+  if product.selected_or_first_available_variant.featured_media
+    assign seoMedia = product.selected_or_first_available_variant.featured_media
+  else
+    assign seoMedia = product.featured_media
```
Comment (sofiamatulis) [inline sections/main-product.liquid:483]:
```suggestion
    assign seo_media = product.featured_media
```

File: sections/main-product.liquid [hunk 4]
```diff
@@ -476,16 +476,24 @@
...
     "@type": "Product",
     "name": {{ product.title | json }},
     "url": {{ shop.url | append: product.url | json }},
-    {%- if product.selected_or_first_available_variant.featured_media -%}
-      {%- assign media_size = product.selected_or_first_available_variant.featured_media.preview_image.width | append: 'x' -%}
+    {%- if seoMedia -%}
+      {%- assign media_size = seoMedia.preview_image.width | append: 'x' -%}
       "image": [
-        {{ product.selected_or_first_available_variant.featured_media | img_url: media_size | prepend: "https:" | json }}
+        {{ seoMedia | img_url: media_size | prepend: "https:" | json }}
```
Comment (sofiamatulis) [inline sections/main-product.liquid:496]:
```suggestion
        {{ seo_media | img_url: media_size | prepend: "https:" | json }}
```

File: sections/featured-product.liquid [hunk 5]
```diff
@@ -402,16 +402,24 @@
   <script src="{{ 'product-model.js' | asset_url }}" defer></script>
 {%- endif -%}
 
+{%- liquid
+  if product.selected_or_first_available_variant.featured_media
+    assign seoMedia = product.selected_or_first_available_variant.featured_media
```
[the remark this answers is not in the document]
Comment (ludoboludo) [reply sections/featured-product.liquid]: 👍  good point. I've updated it to be using underscore instead.

**Review by sofiamatulis — APPROVED** [review]
thanks for the changes 🎉

File: sections/featured-product.liquid [hunk 6]
```diff
@@ -402,16 +402,24 @@
...
+
 <script type="application/ld+json">
   {
     "@context": "http://schema.org/",
     "@type": "Product",
     "name": {{ product.title | json }},
     "url": {{ shop.url | append: product.url | json }},
-    {%- if product.selected_or_first_available_variant.featured_media -%}
-      {%- assign media_size = product.selected_or_first_available_variant.featured_media.preview_image.width | append: 'x' -%}
+    {%- if seo_media -%}
```
Comment (KaichenWang) [inline sections/featured-product.liquid:419]: Not a big deal, but would be good to preserve whitespace inside JSON. For example,

```suggestion
    {% if seo_media -%}
```

**Review by KaichenWang — DISMISSED** [review]
Looks good!

**Review by KaichenWang — APPROVED** [review]

**Review by sofiamatulis — APPROVED** [review]

**Pull request merged**